File: Olympiad/COCI/official/2007/contest4_solutions/prsteni.cpp

/*
  Croatian Open Competition in Informatics 2006/2007
  Contest 4 - Task PRSTENI
  Programming language C++
*/

#include <cstdio>

int gcd( int a, int b ) { return b != 0 ? gcd( b, a%b ) : a; } // najveca zajednicka mjera od a i b

int main(void) {
   int n, first, k;
   scanf( "%d%d", &n, &first );
   for( int i = 0; i < n-1; i++ ) {
      scanf( "%d", &k );
      printf( "%d/%d\n", first/gcd( first, k ), k/gcd( first, k ) );
   }
   return 0;
}

File: Olympiad/COCI/official/2007/contest4_solutions/prsteni.pas

(*
  Croatian Open Competition in Informatics 2006/2007
  Contest 4 - Task PRSTENI
  Programming language Pascal
*)

program prsteni;

function gcd( a, b: integer ):integer;
begin
    if( b = 0 ) then gcd := a
    else gcd := gcd( b, a mod b );
end;

var i, n, first, k: integer;

begin
    read( n, first );
    for i := 1 to n-1 do begin
        read( k );
        writeln( first div gcd( first, k ), '/', k div gcd( first, k ) );
    end;
end.
